### ISSUE TITLE
Allow usage of Ember Data 2.0?

### ISSUE THREAD

**Issue by devinus** [lytics/ember-data-model-fragments#128]
https://github.com/lytics/ember-data.model-fragments/blob/e34d1540cacb5520ec6097682299757fed9c3d02/bower.json#L21

**Comment by slindberg** [comment]
Have you been running v0.4.x successfully against 2.0? My long term plan was to do a v1.13/v2.0 split like ED, which is why the current release is bound to v1.13.x. The problem with loosening the ED version right now is that the v1.13 and v2.0 internals will likely diverge, forcing the v0.4.x minor version to choose which version to be compatible with. Not sure what the best path forward here is, other than pushing forward with #110

**Comment by matthewnessworthy** [comment]
With ember 2.0.2 & ember-data 2.0.0 no warnings are thrown, however, model fragments don't work. From my initial trial the deserialization results in a single shared model fragment filled with default values.
There are also about 34 failing tests after following the testing procedure described in the README.md.
Have you decided on a plan regarding the 1.x/2.x split?

**Comment by slindberg** [comment]
> With ember 2.0.2 & ember-data 2.0.0 no warnings are thrown, however, model fragments don't work.

@matthewnessworthy I'm not overly concerned with this silently failing, since you probably had to add a line to the bower resolutions just to make it stop complaining. This is what dependency versioning is for.

I still don't have a better plan than simply releasing 1.x then branching off 2.x :confused:

**Comment by kmiyashiro** [comment]
@matthewnessworthy my fragments appear to be deserializing fine, I'm using the RESTAdapter.

Ember 2.0.2
ED 2.0.1
Frag 0.4

**Comment by vasilakisfil** [comment]
Also fail to deserialize here: Ember 2.0.2, ED 2.0.1, Frag 0.4.1. But I am using ActiveModel adapter, not sure if it has to do with the issue (https://github.com/ember-data/active-model-adapter).

**Comment by christophermlne** [comment]
I was struggling to pinpoint exactly what my issue was, as I'm a first time user of this project, but after reading this thread, I am sure this is my issue. My fragments serialize fine, but fail to deserialize like the others.

If I create the fragments, push them to the store and associate them to the parent using the console, everything is fine. I can query the parent record in rails and see the fragments have saved correctly.

Hit refresh and the parent record and related models load properly, all except the fragments, which I can see have not made it into the store by looking at the Ember Inspector's data tab.

If I set a breakpoint in a fragment's serializer when I hit reload, I can see the data coming back through the hash variable is fine.

So something is happening between the serializer voodoo and when the record should be pushed to the store...

Wish I knew more about Ember data internals so I could be more helpful.

Looks like the sensible thing to do is downgrade Ember/Ember Data to 1.13.x?

```
DEBUG: -------------------------------
ember.debug.js:5227 DEBUG: Ember           : 2.0.2
ember.debug.js:5227 DEBUG: Ember Data      : 2.0.0
ember.debug.js:5227 DEBUG: jQuery          : 1.11.3
ember.debug.js:5227 DEBUG: Model Fragments : 0.4.1
ember.debug.js:5227 DEBUG: -------------------------------
```

**Comment by slindberg** [comment]
@christophermlne yes, for the time being it's probably best to downgrade to v1.13.x

**Comment by jimmyjamieson** [comment]
Is there a time frame for ED 2.0+ support?

**Comment by elwayman02** [comment]
@slindberg now that 1.13.x is supported, what's left for 2.0 support?

**Comment by slindberg** [comment]
@elwayman02 what's left are the deserialization issues mentioned on this thread.

**Comment by slindberg** [comment]
Released v2.0.0 for compatibility with ED v2.x. As always, please report any issues you find :smile:
